src/handsfree-audio.c

```diff
@@ -69,6 +70,17 @@ static ofono_bool_t has_wideband = FALSE;
 static int defer_setup = 1;
 static ofono_bool_t transparent_sco = FALSE;
 
+static const char* card_type_to_string(enum ofono_handsfree_card_type type)
+{
+	switch (type) {
+	case OFONO_HANDSFREE_CARD_TYPE_HANDSFREE:
+		return "handsfree";
+	case OFONO_HANDSFREE_CARD_TYPE_GATEWAY:
+		return "gateway";
+	}
+	return "";
```

Comment:
All other *_to_string methods in ofono doesn't do that. IMHO it's up to the caller to actually make sure he knows what he has set and this normally shouldn't happen as if no casting is involved only enum values can be set.